src/plugins/data/common/search/search_source/query_to_fields.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataViewLazy } from '@kbn/data-views-plugin/common';
+import { fromKueryExpression, getKqlFieldNames } from '@kbn/es-query';
+import type { SearchRequest } from './fetch';
+import { EsQuerySortValue } from '../..';
+
+export async function queryToFields({
+  dataView,
+  sort,
+  request,
+}: {
+  dataView: DataViewLazy;
+  sort?: EsQuerySortValue | EsQuerySortValue[];
+  request: SearchRequest;
+}) {
+  let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+  if (sort) {
+    const sortArr = Array.isArray(sort) ? sort : [sort];
+    for (const s of sortArr) {
+      const keys = Object.keys(s);
+      fields = fields.concat(keys);
+    }
+  }
+  for (const query of request.query) {
+    if (query.query) {
+      const nodes = fromKueryExpression(query.query);
+      const queryFields = getKqlFieldNames(nodes);
+      fields = fields.concat(queryFields);
+    }
+  }
+  const filters = request.filters;
+  if (filters) {
+    const filtersArr = Array.isArray(filters) ? filters : [filters];
+    for (const f of filtersArr) {
+      fields = fields.concat(f?.meta?.key);
```

Comment:
>fields = fields.concat(f?.meta?.key);

The optional chaining worries me. It was needed to pass the functional test. I'd rather figure out where the filter is being composed and determine whether this is being handled correctly.

Reply:
There are a couple of use cases here to be aware of:

1. Filters coming from the unified search bar
  a. "Regular" filters where a field is specified (where `meta.key` is present)
  b. "Custom" filters ("Edit as query DSL", where `meta.key` is not present but `meta` is)
2. Any other Elasticsearch query DSL filter that gets passed in by consumers (not coming from unified search, and these probably won't have a `meta` key at all)

We should probably also update this logic to filter out filters from unified search bar that are disabled (`meta.disabled === true`) since those aren't needed in the call to `field_caps`.

Reply:
>Any other Elasticsearch query DSL filter that gets passed in by consumers (not coming from unified search, and these probably won't have a meta key at all)

Based on this it sounds like the change is necessary.

Additionally, we should only be appending if `meta.key` exists. And, for completeness, if `meta.disabled !== true`